airbyte-cdk/python/unit_tests/test_entrypoint.py

```diff
@@ -328,94 +328,100 @@ def test_filter_internal_requests(deployment_mode, url, expected_error):
                 stream_state=AirbyteStateBlob(updated_at="2024-02-02")))),
-            {HashableStreamDescriptor(name="customers", namespace="public"): 100},
+            {HashableStreamDescriptor(name="customers", namespace="public"): 100.0},
             AirbyteMessage(type=Type.STATE, state=AirbyteStateMessage(type=AirbyteStateType.STREAM, stream=AirbyteStreamState(
                 stream_descriptor=StreamDescriptor(name="customers", namespace="public"),
                 stream_state=AirbyteStateBlob(updated_at="2024-02-02")), sourceStats=AirbyteStateStats(recordCount=100.0))),
-            {HashableStreamDescriptor(name="customers", namespace="public"): 0},
+            {HashableStreamDescriptor(name="customers", namespace="public"): 0.0},
             id="test_handle_state_message_with_descriptor",
         ),
         pytest.param(
             AirbyteMessage(type=Type.STATE, state=AirbyteStateMessage(type=AirbyteStateType.STREAM, stream=AirbyteStreamState(
                 stream_descriptor=StreamDescriptor(name="others", namespace="public"),
                 stream_state=AirbyteStateBlob(updated_at="2024-02-02")))),
-            {HashableStreamDescriptor(name="customers", namespace="public"): 100},
+            {HashableStreamDescriptor(name="customers", namespace="public"): 100.0},
             AirbyteMessage(type=Type.STATE, state=AirbyteStateMessage(type=AirbyteStateType.STREAM, stream=AirbyteStreamState(
                 stream_descriptor=StreamDescriptor(name="others", namespace="public"),
                 stream_state=AirbyteStateBlob(updated_at="2024-02-02")), sourceStats=AirbyteStateStats(recordCount=0.0))),
-            {HashableStreamDescriptor(name="customers", namespace="public"): 100,
-             HashableStreamDescriptor(name="others", namespace="public"): 0},
+            {HashableStreamDescriptor(name="customers", namespace="public"): 100.0,
+             HashableStreamDescriptor(name="others", namespace="public"): 0.0},
             id="test_handle_state_message_no_records",
         ),
     ]
 )
 def test_handle_record_counts(incoming_message, stream_message_count, expected_message, expected_records_by_stream):
     entrypoint = AirbyteEntrypoint(source=MockSource())
-    actual_record = entrypoint.handle_record_counts(message=incoming_message, stream_message_count=stream_message_count)
-    assert actual_record == expected_message
-    assert stream_message_count == expected_records_by_stream
+    actual_message = entrypoint.handle_record_counts(message=incoming_message, stream_message_count=stream_message_count)
+    assert actual_message == expected_message
+
+    for stream_descriptor, message_count in stream_message_count.items():
+        assert message_count == expected_records_by_stream[stream_descriptor]
+        assert isinstance(message_count, float)  # Python assertions against different number types won't fail if the value is equivalent
```

Comment:
```suggestion
        assert isinstance(message_count, float)
        # Python assertions against different number types won't fail if the value is equivalent
        assert message_count == expected_records_by_stream[stream_descriptor]
```
Small ordering nit to make the comment make more sense